### DISCUSSION TITLE
How to add custom keybinds for telescope

### DISCUSSION THREAD

**Discussion by kyrillh** [LazyVim/LazyVim#809] [Q&A] [answered]
Firstly, thanks for the great setup! I am new to vim/neovim and am really enjoying using LazyVim.
I would like to add a couple of custom keybindings for telescope that enable scrolling through the current list using `<C-j>` and `<C-k>` while in insert mode. I added the following to the lua/config/keymaps.lua file: 
```
vim.keymap.set("i", "<C-k>", require("telescope.actions").move_selection_next, { noremap = true, silent = true })
```
Restarting neovim results in the error 
```
vim/keymap.lua:0: rhs: expected string|function, got table
```
I am guessing the fix is really simple since I have no idea what I am doing.

**Comment by dpetka2001** [comment]
Those are internal keymaps to telescope and you should put them in the spec where you setup telescope. See [docs](https://www.lazyvim.org/plugins/editor#telescopenvim), you should put them in `opts.defaults.mappings`.

**Comment by dpetka2001** [comment, marked as answer]
```lua
return {
  {
    "nvim-telescope/telescope.nvim",
    opts = {
      defaults = {
        mappings = {
          i = {
            ["<C-k>"] = "move_selection_next",
            ["<C-j>"] = "move_selection_previous",
          },
        },
      },
    },
  },
}
```
Something like this should work. Put it in a file in `~/.config/nvim/lua/plugins` and you should be good to go.

**Reply by kyrillh** [reply]
Thanks! Your first solution worked as well, but I prefer this solution as it only makes changes in the ~/.config/nvim directory.
A side question: why does this not override the existing telescope mappings?

**Reply by dpetka2001** [reply]
I think it just sets new keybindings for the actions you want. The old keybindings still remain. Except if you explicitly specify `<C-n>/<C-p>` to some other actions or disable them.

**Reply by tmcw** [reply]
if I wanted to override or disable a default mapping… how would I do that? I've been trying a bunch of different approaches - removing the mapping in `keymaps.lua`, changing this `opts` thing to a method as suggested in the docs, and haven't figured out the trick.

What I'd really want to do is restore the "control-tab" binding in Telescope so that I can open a result in a new tab instead of opening all the results in trouble.nvim. This is the latest thing I've tried, but I've also tried this, but with the `telescope.actions` new_tab command where `false` is.

```
    opts = {
      defaults = {
        layout_strategy = "horizontal",
        layout_config = { prompt_position = "top" },
        sorting_strategy = "ascending",
        winblend = 0,
        mappings = {
          i = {
            ["<c-t>"] = false,
          },
        },
      },
    },
```

**Reply by jyuan0** [reply]
What you have should work. Maybe there's a typo somewhere? For reference:
```
return {
  {
    "nvim-telescope/telescope.nvim",
    opts = {
      defaults = {
        mappings = {
          i = {
            ["<c-t>"] = require("telescope.actions").select_tab,
          },
        },
      },
    },
  },
}
```

**Reply by dpetka2001** [reply]
`<C-t>` is not "Control+Tab" actually. It's "Control+t" (lowercase T). Maybe you're testing the wrong thing?

**Reply by tmcw** [reply]
I was _really_ testing the wrong thing, I had forgotten to remove the early `return {}` from the example.lua file 🤦 . Thanks for the help, sorry for the false alarm, another Monday…

**Reply by jyuan0** [reply]
Duh, thank you @dpetka2001.

**Comment by shu7bh** [comment]


**Reply by dpetka2001** [reply]
Have you seen how LazyVim spec files are structured? This isn't a correct LazyVim file (at least from what you showed). Just look at the example from the accepted answer and you will see how it has to be structured. And also plz read the docs to understand how it works.
